File: Lua/LongestLines.lua
lines = {}
for line in io.lines(arg[1]) do
  lines[#lines + 1] = line
end
N = table.remove(lines, 1)
table.sort(lines, function(a, b) return #a > #b end)
for i = 1,N do
  print(lines[i])
end

File: Python3/LongestLines.py
import sys
from itertools import islice

with open(sys.argv[1], 'r') as test_cases:
    lines = [test for test in test_cases]
    N = int(lines[0])
    top_N = islice(
        reversed(sorted(lines[1:], key=len)),
        N
    )
    for line in top_N:
        print(line)


